torch/csrc/autograd/python_function.cpp

```diff
@@ -201,6 +202,64 @@ auto PyNode::name() const -> std::string {
   return name;
 }
 
+#ifdef COMPILED_AUTOGRAD
+void PyNode::compiled_args(CompiledNodeArgs& args) {
```

Comment:
You're not taking out the GIL here. Is this OK? These are virtual methods defined on a base C++ non-Python class.

Reply:
GIL is held for all of compiled_autograd:
https://github.com/pytorch/pytorch/blob/d220273d466a1c1f20ab5f211c942dc9c2723f27/torch/csrc/dynamo/compiled_autograd_py.cpp#L287

We are making lots of calls into python and not actually doing any compute, so seems better to just take it once.